edx_exams/apps/core/api.py

```diff
@@ -41,6 +41,17 @@ def get_latest_attempt_for_user(user_id):
     return latest_attempt
 
 
+def get_attempt_by_attempt_number_and_resource_id(attempt_number, resource_id):
```

Comment:
it looks like we have a very similar function already for `get_attempt_for_user_with_attempt_number_and_resource_id` any way we can cut down on duplicate code paths here?